### PR TITLE
TMEDIA-684 - Grid Component

### PR DESCRIPTION
**Pull request by matthewroach**
## Ticket

- [TMEDIA-684](https://arcpublishing.atlassian.net/browse/TMEDIA-684)

## Description

Add `Grid` component to allow grid layout to be controlled via enhanced styling

## Test Steps

1. Checkout branch - `git checkout TMEDIA-684-grid`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the button storybook documentation

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

### CODE REVIEW

File: src/components/grid/_index.scss [hunk 1]
```diff
@@ -0,0 +1,9 @@
+@use "../../scss";
+
+.c-grid {
+	--grid-template-columns: var(--c-grid-template-columns, repeat(auto-fit, minmax(15em, 1fr)));
```
Comment (nschubach) [inline src/components/grid/_index.scss:4]: My only concern here is the min-width.  I grabbed that value from: https://codepen.io/web-dot-dev/pen/QWMMVPm so if that sort of minimum width value (15em ~= 240px) makes sense, then we can stick with it.
image

15em breaks to single column on the iPhone 12:
image

10em breaks to two:
image

I don't know that I like 2 columns on such a small device, but the mobile design is 2:
image

Comment (nschubach) [reply src/components/grid/_index.scss:4]: depending on gap, 32% actually defaults to 2 column on mobile and broadens (after ~832px) to 3 columns on everything else.

Comment (matthewroach) [reply src/components/grid/_index.scss:4]: I played around with the min value and came to the conclusion `15em` being ~240px as the min width of a grid children item seemed to be a sane default

**Review by nschubach — APPROVED** [review]
Seems to do everything we need

**Pull request merged**